internal/testcase/ecs_pod.go

```diff
@@ -1 +1,165 @@
+				[]cocoa.ECSContainerDefinition{*containerDef.SetEnvironmentVariables(
+					[]cocoa.EnvironmentVariable{*secret, *secretOwned})})
+
+			p, err := pc.CreatePod(ctx, optsSecret)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Running, info.Status)
+			assert.Len(t, info.Resources.Secrets, 2)
+
+			require.NoError(t, p.Stop(ctx))
+
+			info, err = p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Stopped, info.Status)
+			require.Len(t, info.Resources.Secrets, 2)
+
+			arn := info.Resources.Secrets[0].Name
+			id, err := v.GetValue(ctx, *arn)
+			require.NoError(t, err)
+			require.NotNil(t, id)
+		},
+		"StopFailsOnIncorrectPodStatus": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, opts)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			require.NoError(t, p.Stop(ctx))
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotZero(t, info)
+			assert.Equal(t, cocoa.Stopped, info.Status)
+
+			require.Error(t, p.Stop(ctx))
+		},
+		"DeleteSucceeds": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			p, err := pc.CreatePod(ctx, opts)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Running, info.Status)
+
+			require.NoError(t, p.Delete(ctx))
+
+			info, err = p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Deleted, info.Status)
+		},
+		"DeleteSucceedsWithSecrets": func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator) {
+			secret := cocoa.NewEnvironmentVariable().SetName("secret1").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName(testutil.NewSecretName("name1")).SetValue("value1"))
+			secretOwned := cocoa.NewEnvironmentVariable().SetName("secret2").
+				SetSecretOptions(*cocoa.NewSecretOptions().SetName(testutil.NewSecretName("name2")).SetValue("value2"))
+			secretOwned.SecretOpts.SetOwned(true)
+
+			optsSecret := optsSecret.SetContainerDefinitions(
+				[]cocoa.ECSContainerDefinition{*containerDef.SetEnvironmentVariables(
+					[]cocoa.EnvironmentVariable{*secret, *secretOwned})})
+
+			p, err := pc.CreatePod(ctx, optsSecret)
+			require.NoError(t, err)
+			require.NotZero(t, p)
+
+			info, err := p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Running, info.Status)
+			require.Len(t, info.Resources.Secrets, 2)
+
+			require.NoError(t, p.Delete(ctx))
+
+			info, err = p.Info(ctx)
+			require.NoError(t, err)
+			require.NotNil(t, info)
+			assert.Equal(t, cocoa.Deleted, info.Status)
+			require.Len(t, info.Resources.Secrets, 2)
+
+			arn0 := info.Resources.Secrets[0].Name
+			val, err := v.GetValue(ctx, *arn0)
```

Comment:
Just to check my understanding of this test - even if you've already deleted the secret with `ForceDeleteWithoutRecovery`, it'll still return the value to you?

Reply:
No, it shouldn't because the secret is already scheduled for deletion. However, in this case, the first secret (`secret`) is not owned by the pod, so it doesn't get deleted, which is why I check that there is no error and the value that getting the secret's value returns us is the one we set it as in the beginning. For the second secret (`secretOwned`), since its `Owned` field is set as true, it does get deleted and returns an error when we try to get its value.